llvm/lib/ObjectYAML/GOFFEmitter.cpp

```diff
@@ -73,6 +73,22 @@ raw_ostream &operator<<(raw_ostream &OS, const yaml::BinaryRef &Data) {
 // physical records are created for the data. Possible fill bytes at the end of
 // a physical record are written automatically.
 class GOFFOstream : public raw_ostream {
+
+public:
+  explicit GOFFOstream(raw_ostream &OS)
+      : OS(OS), LogicalRecords(0), RemainingSize(0), NewLogicalRecord(false) {
+    SetBufferSize(GOFF::PayloadLength);
+  }
+
+  ~GOFFOstream() { finalize(); }
+
+  void newRecord(GOFF::RecordType Type, size_t Size);
```

Comment:
Nit: this should be a verb phrase, e.g. `makeNewRecord`. See https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly.